subsys/net/ip/ipv6_fragment.c

```diff
@@ -432,24 +432,23 @@ static int shift_packets(struct net_ipv6_reassembly *reass, int pos)
 			NET_DBG("Moving [%d] %p (offset 0x%x) to [%d]",
```

Comment:
The `net: ipv6_fragment: fix shift_packets() algorithm` is also a great candidate to create a unit test. It should be quite straight forward to unit test.